src/utils.cpp

```diff
@@ -176,36 +217,16 @@ Term utils::gen_bw(const Op op, const uint64_t bv_width, uint64_t granularity, c
   }
   uint64_t num_of_blocks = bv_width / block_size;
   if (opts.use_sum_bvops) {
-    Term sum = int_zero_;
-    for (uint64_t i = 0; i < num_of_blocks; i++) {
-      Term block = gen_block(op, a, b, i, block_size, side_effects);
-      Term power_of_two = pow2(i);
-      Term sum_part = solver_->make_term(Mult, block, power_of_two);
-      sum = solver_->make_term(Plus, sum, sum_part);
-    }
-    return sum;
+    Term sum = gen_bw_sum(op, bv_width, granularity, a, b, side_effects);
+    // add equality to extra assertions
+    side_effects.push_back(solver_->make_term(Equal, sigma, sum));
   } else {
     Sort intsort = solver_->make_sort(INT);
```

Comment:
Ideally this branch will call a `gen_bw_bit_equations` or something, that would include the code below.

Reply:
Oh good idea! Added with 45b619676518b9c87b9f513e14f08f24cc87a0c2